qiskit/algorithms/gradients/base_estimator_gradient.py

```diff
@@ -0,0 +1,122 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Abstract Base class of Gradient for Estimator.
+"""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.exceptions import QiskitError
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .estimator_gradient_result import EstimatorGradientResult
+
+
+class BaseEstimatorGradient(ABC):
+    """Base class for an EstimatorGradient to compute the gradients of the expectation value."""
+
+    def __init__(
+        self,
+        estimator: BaseEstimator,
+        **run_options,
+    ):
+        """
+        Args:
+            estimator: The estimator used to compute the gradients.
+        """
+        self._estimator: BaseEstimator = estimator
+        self._circuits: Sequence[QuantumCircuit] = []
+        self._circuit_ids: dict[int, int] = {}
+        self._default_run_options = run_options
+
+    def evaluate(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        observables: Sequence[BaseOperator | PauliSumOp],
+        parameter_values: Sequence[Sequence[float]],
+        partial: Sequence[Sequence[Parameter]] | None = None,
+        **run_options,
+    ) -> EstimatorGradientResult:
+        """Run the job of the gradients of expectation values.
+
+        Args:
+            circuits: The list of quantum circuits to compute the gradients.
+            observables: The list of observables.
+            parameter_values: The list of parameter values to be bound to the circuit.
+            partial: The list of Parameters to calculate only the gradients of the specified parameters.
```

Comment:
I guess when I see `partial` I think partial functions in relation to Python, Could this simply be named parameters, as it you specify which parameters the gradient is computed over. And although this says `list of Parameters` this is actually a list of lists (Sequence of Sequence) where presumably each list in the list corresponds to each circuit in the list. I guess there is no way to simply say for some circuits I want to be specific but for others use them all - well you can but you will need to list them all since its not Sequence[Sequence[Parameter] | None] where None there would be on a circuit by circuit basis.

Reply:
hmm, naming `parameters` makes sense. I will change it. Also, `parameters: Sequence[Sequence[Parameter] | None] | None = None` looks better and user-friendly for multiple circuits. Users can specify parameters like `parameters=[[parameter1], None, [parameter2]]` for 3 circuits, and use all parameters for circuit2.